### ISSUE TITLE
Bug: [QueryBuilder] if set limit to 0, it will return all data. Perhaps it should return 0 data.

### ISSUE THREAD

**Issue by RikhzanAmir** [codeigniter4/CodeIgniter4#8278]
### PHP Version

8.2

### CodeIgniter4 Version

4

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

Linux

### Which server did you use?

fpm-fcgi

### Database

MariaDB 10.2.44

### What happened?

`$builder->limit(0, 0)` , set the limit => 0 and the query will shows all the table rows. It should return no rows because limit is 0.

### Steps to Reproduce

```php
$builder->limit(0,0)
```

### Expected Output

it should return no rows if it set limit to 0.

**Comment by kenjis** [comment]
Thank you for reporting. This is very interesting.

> $builder->limit(10, 20);
// Produces: LIMIT 20, 10 (in MySQL. Other databases have slightly different syntax)
https://codeigniter4.github.io/CodeIgniter4/database/query_builder.html#limit

As I read it, I expect `limit(0, 0)` produces `LIMIT 0, 0`. 
In reality, however, no LIMIT clause is generated.

**Comment by kenjis** [comment]
This behavior has not changed since the beginning of 4.0.
https://github.com/codeigniter4/CodeIgniter4/blob/4d2746f433c8d375a3ed04207967336839c56a3f/system/Database/BuilderTrait.php#L2400-L2404

**Comment by kenjis** [comment]
In CI3, the following code generates `SELECT *
FROM "job"
 LIMIT 0` (sqlite3):
```php
		$jobs = $this->db->limit(0, 0)
		                      ->get('job')
		                      ->result_array();
```

**Comment by RikhzanAmir** [comment]
I am writing this query in MariaDB:  "SELECT * FROM payment limit 0 offset 0"  , and it shows no result.

So on Ci3, and Ci4 I am expecting $builder->limit(0,0)  or $builder->limit($set, $offset)  do the same. 

Maybe, it is not a bug,  some query do different depending on database we use.

https://dev.mysql.com/doc/refman/8.0/en/limit-optimization.html

"LIMIT 0 quickly returns an empty set. This can be useful for checking the validity of a query. It can also be employed to obtain the types of the result columns within applications that use a MySQL API that makes result set metadata available."

**Comment by kenjis** [comment]
Do you have a use case for limit 0 with Query Builder?

`LIMIT` is not in SQL standards. So we don't need to implement like MySQL's.
But since "limit 1" returns 1 row, it is logical that "limit 0" would return 0 rows. 
However, the implementation will be more complicated than it is now.

**Comment by kenjis** [comment]
```php
if ($quantity === 0) {
    return [];
}
$query = $this->db->table('order')->limit($quantity)->offset($offset)->get()->getResultArray();
```

**Comment by kenjis** [comment]
I sent PR #8280 to fix this.

**Comment by kenjis** [comment]
You can fix this behavior in v4.5.0. See #8280
